From 7dcbee530800b16534d0f3e8db375492c03b3e0a Mon Sep 17 00:00:00 2001
From: lijuang <lijuang@codeaurora.org>
Date: Tue, 10 Oct 2017 10:49:39 +0800
Subject: QcomModulePkg: Add return value check after AllocatePages

Exit the function if it's failed to allocate pages
for memory map

Change-Id: I7c6bf5378d78f092a21a2d6b55799563a061cfd7
---
 QcomModulePkg/Library/BootLib/ShutdownServices.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/QcomModulePkg/Library/BootLib/ShutdownServices.c b/QcomModulePkg/Library/BootLib/ShutdownServices.c
index 8f0faa6..2cee72c 100644
--- a/QcomModulePkg/Library/BootLib/ShutdownServices.c
+++ b/QcomModulePkg/Library/BootLib/ShutdownServices.c
@@ -53,6 +53,11 @@ EFI_STATUS ShutdownUefiBootServices (VOID)
 
 			Pages = EFI_SIZE_TO_PAGES (MemoryMapSize) + 1;
 			MemoryMap = AllocatePages (Pages);
+            if (!MemoryMap) {
+                DEBUG ((EFI_D_ERROR,
+                    "Failed to allocate pages for memory map\n"));
+                return EFI_OUT_OF_RESOURCES;
+            }
 
 			//
 			// Get System MemoryMap
-- 
cgit v1.1

